torch/csrc/jit/aten_schema.cpp

```diff
@@ -0,0 +1,251 @@
+          int64_t v = std::stoll(n);
+          return at::full({}, v, at::kLong);
+        }
+    }
+  }
+  at::Tensor parseConstantList(TypeKind kind) {
+    auto tok = L.expect('[');
+    std::vector<at::Tensor> vs;
+    if(L.cur().kind != ']') {
+      do {
+        vs.push_back(parseSingleConstant(kind));
+      } while(L.nextIf(','));
+    }
+    L.expect(']');
+    if(vs.size() == 0) {
+      switch(kind) {
+        case TypeKind::FloatType:
+          return at::empty({}, at::kFloat);
+        case TypeKind::IntType:
+          return at::empty({}, at::kLong);
+        default:
+          throw ErrorReport(tok) << "empty lists are only supported for float or int types.";
+      }
+    }
+    return at::stack(vs);
+  }
+  at::Tensor parseTensorDefault(const SourceRange& range) {
+    if("None" == L.expect(TK_IDENT).text()) {
+      return at::Tensor();
+    } else {
+      throw ErrorReport(range) << "invalid tensor default value";
+    }
+  }
+  void parseDefaultValue(Argument& arg) {
+    auto range = L.cur().range;
+    switch(arg.type->kind()) {
+      case TypeKind::DynamicType: {
+        arg.default_value = parseTensorDefault(range);
+      }  break;
+      case TypeKind::NumberType:
+      case TypeKind::IntType:
+      case TypeKind::FloatType:
+        arg.default_value = parseSingleConstant(arg.type->kind());
+        break;
+      case TypeKind::ListType: {
+        auto elem_kind = arg.type->cast<ListType>()->getElementType();
+        if(L.cur().kind == TK_IDENT) {
+          arg.default_value = parseTensorDefault(range);
+        } else if(arg.N && L.cur().kind != '[') {
+          arg.default_value = parseSingleConstant(elem_kind->kind()).expand({*arg.N});
+        } else {
+          arg.default_value = parseConstantList(elem_kind->kind());
+        }
+      } break;
+      default:
+        throw ErrorReport(range) << "unexpected type, file a bug report";
+    }
+  }
+
+  template<typename T>
+  void parseList(int begin, int sep, int end, std::vector<T>& result, void (SchemaParser::*parse)(std::vector<T>&)) {
+    auto r = L.cur().range;
+    if (begin != TK_NOTHING)
+      L.expect(begin);
+    if (L.cur().kind != end) {
+      do {
+        (this->*parse)(result);
+      } while (L.nextIf(sep));
+    }
+    if (end != TK_NOTHING)
+      L.expect(end);
+  }
+  Lexer L;
+  bool kwarg_only;
+  at::Tensor one = at::full({}, 1, at::kLong);
+  at::Tensor zero = at::full({}, 0, at::kLong);
+};
+}
+
+using SchemaMap = std::unordered_map<std::string, std::vector<FunctionSchema>>;
+
+// defined in aten_schema_declarations.cpp
+extern const char * schema_declarations;
+
+std::vector<FunctionSchema> createOperatorSchemas() {
+  return script::SchemaParser(schema_declarations).parseDeclarations();
+}
+
+std::vector<FunctionSchema> & getOperatorSchemas() {
+  static std::vector<FunctionSchema> schema = createOperatorSchemas();
+  return schema;
+}
+
+static SchemaMap createSchemaMap() {
+  auto& schemas = getOperatorSchemas();
```

Comment:
Not for this PR, but we should have some "mini-benchmark" that checks how long this takes, and fails if it takes too long to load all of the schemas :)